functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/jobs/JobActivityHistorySpec.groovy

```diff
@@ -0,0 +1,97 @@
+
+package org.rundeck.tests.functional.selenium.jobs
+
+import org.openqa.selenium.support.ui.WebDriverWait
+import org.rundeck.util.annotations.SeleniumCoreTest
+import org.rundeck.util.container.SeleniumBase
+import org.rundeck.util.gui.pages.login.LoginPage
+import org.rundeck.util.gui.pages.execution.ExecutionShowPage
+import org.rundeck.util.gui.pages.jobs.JobListPage
+import org.rundeck.util.gui.pages.activity.ActivityPage
+import org.rundeck.util.common.jobs.JobUtils
+import java.time.Duration
+
+@SeleniumCoreTest
+class JobActivityHistorySpec extends SeleniumBase {
+    WebDriverWait wait
+    String jobId
+
+    def setupSpec() {
+        setupProject(SELENIUM_BASIC_PROJECT)
+    }
+
+    /**
+     * Setup the test environment.
+     */
+    def setup() {
+        // Set up WebDriverWait to wait up to 30 seconds for elements that take time to appear or change
+        wait = new WebDriverWait(getDriver(), Duration.ofSeconds(30))
+        (go LoginPage).login(TEST_USER, TEST_PASS)
+        def jobDefinition = JobUtils.generateScheduledExecutionXml("TestJobActivityHistory")
+        def client = getClient()
+        def response = JobUtils.createJob(SELENIUM_BASIC_PROJECT, jobDefinition, client)
+        assert response.succeeded.size() > 0: "Failed to create job"
+        jobId = response.succeeded[0].id
+        def runResponse = JobUtils.executeJob(jobId, client)
+        assert runResponse.code() == 200: "Failed to execute job"
+
+    }
+
+
+    def "review job activity history from Job List Page"() {
+        when: "Navigate to the Job List Page"
+        def activityPage = go ActivityPage, SELENIUM_BASIC_PROJECT
+        activityPage.loadActivityPageForProject(SELENIUM_BASIC_PROJECT)
+
+        then: "Validate job execution is listed in Activity History"
+        activityPage.validatePage()
+        wait.until {
+            def activityList = activityPage.getActivityRows()
+            !activityList.isEmpty()
+        }
+
+        def firstActivityRow = activityPage.getActivityRows().get(0)
```

Comment:
Unnecessary validation:
Your test does not need to check if the execution has succeeded. 
You are testing "activity", not job execution status. Seeing a row is sufficient.
Contrary to the initial gut feel, "less is more" with unit testing. Keep them focused.